kolibri/core/analytics/utils.py

```diff
@@ -62,6 +75,128 @@
 ]
 
 
+def standard_deviation(data):
+    if data:
+        # calculate mean
+        m = sum(data) / len(data)
+        # return population std
+        return round(math.sqrt(sum((d - m) ** 2 for d in data) / len(data)), 2)
+    return None
+
+
+def calculate_demographic_stats(dataset_id=None, channel_id=None, learner=True):
+    stats = {}
+    roles_filter = Q(roles__isnull=False)
+    if learner:
+        roles_filter = Q(roles__isnull=True)
+    queryset = FacilityUser.objects.filter(roles_filter)
+    # handle stats at facility level
+    if dataset_id:
+        queryset = queryset.filter(dataset_id=dataset_id)
+    # handle stats at channel level
+    if channel_id:
+        user_ids = (
+            queryset.filter(contentsummarylog__channel_id=channel_id)
+            .distinct()
+            .values_list("id", flat=True)
+        )
+        # pass distinct user_ids as subquery
+        queryset = FacilityUser.objects.filter(id__in=user_ids)
```

Comment:
What happened when you did this without the subquery? (i.e. just filtering `queryset` directly)

Reply:
Before I was using `distinct`, because filtering by the related `ContentSummaryLog` would return redundant user models, which was also throwing the queries below off because it would distinct the birth years, so this is the workaround to getting unique users.